tests/test_wheels.py

```diff
@@ -0,0 +1,88 @@
+import io
+import textwrap
+
+import pytest
+
+from installer.wheels import SuperfulousRecordColumnsWarning, parse_record_file
+
+
+def test_parse_wheel_record_simple():
+    record_content = textwrap.dedent(
+        """\
+        file.py,sha256=AVTFPZpEKzuHr7OvQZmhaU3LvwKz06AJw8mT\\_pNh2yI,3144
+        distribution-1.0.dist-info/RECORD,,
+        """
+    )
+    records = list(parse_record_file(io.StringIO(record_content)))
```

Comment:
If the function is going to take an iterator of strings, then the example text can just be the `iter()` of a list of lines. That might also make it easier to work with from a `pytest.mark.parametrize()` perspective.

Reply:
Think about this a bit further, maybe it’s wrong to accept an iterable as argument, because this precludes the possibility to use this in an async generator. Maybe it should work like this instead:

```python
def get_record_file_parser():
    ...

parser = get_record_file_parser()
for line in lines:  # Can be async for as well.
   record = parser.send(line)
   # Do something with record...
```

And then we can have a synchronous wrapper of this to accept a synchronous interable.

Reply:
Scratch that, `csv.reader()` is not line-by-line, so `send()` wouldn’t work.